lib/active_record/patches/sqlite3_adapter.rb

```diff
@@ -0,0 +1,46 @@
+module ActiveRecord
+  module Patches
+    module SQLite3Adapter
+      def initialize(...)
+        # version 3.37 is required for strict typing support and the newest json operators
+        raise Litestack::InsufficientSQLiteVersionError if ::SQLite3::SQLITE_VERSION_NUMBER < 3037000
+        
+        super
+        
+        configure_connection unless options.fetch(:noinit, false)
+        
+        ::SQLite3::Database.prepend Litedb
+        ::SQLite3::Statement.prepend Litedb::Statement
+      end
+      
+      private
+      
+      def configure_connection
```

Comment:
I haven't tested that, but being prepended it won't call the original `configure_connection`, right? If so, it would be good to call it so it loads up [Rails' defaults as well](https://github.com/rails/rails/blob/8c23eaa4eb328e4d007507cee0cc4a46a3e803f0/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L694C1-L698).

Reply:
I honestly named this method without checking if it conflicted with a Rails method in the prepended scope. Good catch. I think I'll rename the method to avoid collisions